src/include/parser/parse_func.h

```diff
@@ -75,10 +75,11 @@ extern void check_srf_call_placement(ParseState *pstate, Node *last_srf,
  * Hook interface to select a function from candidates
  */
 typedef FuncCandidateList (*func_select_candidate_hook_type) (int nargs, Oid *input_typeids, FuncCandidateList candidates, bool unknowns_resolved);
+typedef FuncCandidateList (*func_select_candidate_for_special_func_hook_type) (List *names, int nargs, Oid *input_typeids, FuncCandidateList candidates);
```

Comment:
We can possibly reuse the existing `func_select_candidate_hook` hook for the special functions as well isn't it? Just need to update the hook function to have an extra boolean argument to distinguish between normal and special case.

Reply:
+1

Reply:
Done